apps/otlp.go

```diff
@@ -42,45 +59,54 @@ func (r ReceiverOTLP) Type() string {
-		return fmt.Sprintf(`set(%s, %s) where %s != nil and resource.attributes["cloud.platform"] == "gcp_compute_engine"`,
-			target, source, source)
-	}
-	return []otel.Component{
-		otel.GCPResourceDetector(false),
-		{
-			Type: "transform",
-			Config: map[string]interface{}{
-				"error_mode": "ignore",
-				"metric_statements": []map[string]interface{}{{
-					"context": "datapoint",
-					"statements": []string{
-						// location = cloud.availability_zone
-						stmt(`attributes["location"]`, `resource.attributes["cloud.availability_zone"]`),
-						// namespace = host.id
-						stmt(`attributes["namespace"]`, `resource.attributes["host.id"]`),
-						// cluster = "__gce__"
-						stmt(`attributes["cluster"]`, `"__gce__"`),
-						// instance_name = host.name
-						stmt(`attributes["instance_name"]`, `resource.attributes["host.name"]`),
-						// machine_type = host.type
-						stmt(`attributes["machine_type"]`, `resource.attributes["host.type"]`),
-						// job and instance should be automatically set from service labels
-					},
-				}},
+	bmsResource := platform.FromContext(ctx).ResourceOverride
+	components := func(attributes map[string]string, processor otel.Component) []otel.Component {
+		stmt := func(target, source string) string {
+			platformAttr := "gcp_compute_engine"
+			if bmsResource != nil {
+				platformAttr = "gcp_bare_metal_solution"
+			}
+			return fmt.Sprintf(`set(%s, %s) where %s != nil and resource.attributes["cloud.platform"] == "%s"`,
+				target, source, source, platformAttr)
+		}
+		statements := []string{}
+		keys := make([]string, 0, len(statements))
+		for k := range attributes {
+			keys = append(keys, k)
+		}
+		sort.Strings(keys)
+		for _, k := range keys {
+			statements = append(statements, stmt(k, attributes[k]))
+		}
+		return []otel.Component{
+			processor,
+			{
+				Type: "transform",
+				Config: map[string]interface{}{
+					"error_mode": "ignore",
+					"metric_statements": []map[string]interface{}{{
+						"context":    "datapoint",
+						"statements": statements,
+					}},
+				},
 			},
-		},
-		// TODO: Can we just set resource.attributes instead of setting metric attributes and then grouping?
-		otel.GroupByGMPAttrs(),
+			// TODO: Can we just set resource.attributes instead of setting metric attributes and then grouping?
+			otel.GroupByGMPAttrs(),
+		}
+	}
+
+	if bmsResource != nil {
+		return components(bmsAttributes, otel.ResourceTransform(bmsResource.OTelResourceAttributes()))
 	}
+	return components(gceAttributes, otel.GCPResourceDetector(false))
 }
 
-func (r ReceiverOTLP) metricsProcessors() (otel.ExporterType, otel.ResourceDetectionMode, []otel.Component) {
+func (r ReceiverOTLP) metricsProcessors(ctx context.Context) (otel.ExporterType, otel.ResourceDetectionMode, []otel.Component) {
 	if r.MetricsMode != "googlecloudmonitoring" {
-		return otel.GMP, otel.None, r.gmpResourceProcessors()
+		p := r.gmpResourceProcessors(ctx)
```

Comment:
Why did this change from one line to two?

Reply:
This is a leftover from testing.